src/composer.ts

```diff
@@ -658,6 +659,7 @@ export class AtomicTransactionComposer {
             methodResult.rawReturnValue
           );
         }
+        methodResult.txInfo = pendingInfo;
```

Comment:
Can you put this line immediately after `pendingInfo` is created? The reason being an error might be thrown between its current placement and the assignment of `pendingInfo`, meaning the caller would not receive `txInfo` when they should.